frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -28,141 +29,195 @@ import { DefaultCurrencyRotator } from '../../../components/rates/rates';
+  const mobileHeight: number = 150;
+  const isMobile = window.innerWidth <= 640;
+  const hasData = data.chartDataDaily && data.chartDataDaily.length > 0;
+  const hasHourlyData = data.chartDataHourly && data.chartDataHourly.length > 0;
+
+  const { t, i18n } = useTranslation();
+  const { chartDisplay, setChartDisplay } = useContext(AppContext);
+
+  const ref = useRef<HTMLDivElement>(null);
+  const refToolTip = useRef<HTMLSpanElement>(null);
+  const chart = useRef<IChartApi>();
+  const lineSeries = useRef<ISeriesApi<'Area'>>();
+  const formattedData = useRef<FormattedData>({});
+
+  const [source, setSource] = useState<'daily' | 'hourly'>('daily');
+  const [difference, setDifference] = useState<number>();
+  const [diffSince, setDiffSince] = useState<string>();
+  const [tooltipData, setTooltipData] = useState<{
+    toolTipVisible: boolean;
+    toolTipValue?: string;
+    toolTipTop: number;
+    toolTipLeft: number;
+    toolTipTime: number;
+  }>({
     toolTipVisible: false,
-    toolTipValue: undefined,
     toolTipTop: 0,
     toolTipLeft: 0,
     toolTipTime: 0,
-    isMobile: false,
-  };
+  });
+
+  const prevChartDataDaily = usePrevious(data.chartDataDaily);
+  const prevChartDataHourly = usePrevious(data.chartDataHourly);
+  const prevChartFiat = usePrevious(data.chartFiat);
+  const prevHideAmounts = usePrevious(hideAmounts);
 
-  public componentDidMount() {
-    this.createChart();
+  if (source === 'daily' && prevChartDataDaily?.length !== data.chartDataDaily.length) {
+    lineSeries.current?.setData(data.chartDataDaily);
+    chart.current?.timeScale().fitContent();
+    setFormattedData(data.chartDataDaily);
   }
 
-  public componentWillUnmount() {
-    window.removeEventListener('resize', this.onResize);
-    if (this.chart) {
-      this.chart.timeScale().unsubscribeVisibleLogicalRangeChange(this.calculateChange);
-      this.chart.unsubscribeCrosshairMove(this.handleCrosshair);
-    }
+  if (source === 'hourly' && prevChartDataHourly?.length !== data.chartDataHourly.length) {
+    lineSeries.current?.setData(data.chartDataHourly);
+    chart.current?.timeScale().fitContent();
+    setFormattedData(data.chartDataHourly);
   }
 
-  public componentDidUpdate(prev: Props) {
-    const { chartDataDaily, chartDataHourly } = this.props.data;
-    if (!this.chart) {
-      this.createChart();
+  if (prevChartFiat !== data.chartFiat) {
+    reinitializeChart();
+  }
+
+  if (prevHideAmounts !== hideAmounts) {
+    chart.current?.applyOptions({
+      leftPriceScale: {
+        visible: hideAmounts ? false : !isMobile,
+      }
+    });
+  }
+
+  useEffect(() => {
+    initChart();
+    return () => {
+      window.removeEventListener('resize', onResize);
+      if (chart.current) {
+        chart.current.timeScale().unsubscribeVisibleLogicalRangeChange(calculateChange);
+        chart.current.unsubscribeCrosshairMove(handleCrosshair);
+      }
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
is this needed?

Reply:
we explicitly want to run the effect **once** which is possible by passing an empty dependency array according to the react docs. however eslint wants us to either add the used dependencies, or remove the dependency array (which would cause it to run every render). 

so we either ignore the warning with the comment or remove the dependency array but check if it already ran once e.g
```
const chartInitialized = useRef(false);

useEffect..
  if(!chartInitialized) ...chartInitialized = true, initChart() return cleanup
  else do nothing
 ```
 
 so the function in use effect would return `() => void | undefined` not sure if thats even possible ? can it have _no cleanup function_ when some condition is not met?

Reply:
tldr; yes otherwise we get eslint errors